dimod/discrete/cppdqm.pxd

```diff
@@ -0,0 +1,97 @@
+# distutils: language = c++
+# cython: language_level=3
+#
+# Copyright 2019 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# =============================================================================
+
+from libcpp cimport bool
+from libcpp.pair cimport pair
+from libcpp.vector cimport vector
+
+cimport numpy as np
+
+from dimod.bqm.common cimport VarIndex, Bias
+from dimod.bqm.cppbqm cimport AdjVectorBQM as cppAdjVectorBQM
+
+cdef extern from "dimod/adjvectordqm.h" namespace "dimod" nogil:
+
+    cdef cppclass AdjVectorDQM[V, B]:
+        ctypedef V variable_type
+        ctypedef B bias_type
+        ctypedef size_t size_type
+
+	cdef cppAdjVectorBQM[variable_type, bias_type] bqm_ 
+        vector[variable_type] case_starts_
+        vector[vector[variable_type]] adj_
+
+        # constructors
+        # cython cannot handle templated constructors, so we call out the types
+        # explicitly
+
+        AdjVectorDQM() except +
+
```

Comment:
Do I need copy constructors or anything like that ? 
The dqm object does not contain any pointers so shouldnt the default copy constructor work fine ? Since no deep copying is not needed.

Reply:
Probably better to be explicit